ts/screens/preferences/PreferencesScreen.tsx

```diff
@@ -46,33 +41,21 @@ type OwnProps = Readonly<{
 }>;
 
 type Props = OwnProps &
-  LightModalContextInterface &
   ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps> &
   ReduxProps;
 
 type State = {
   isFingerprintAvailable: boolean;
   hasCalendarPermission: boolean;
+  checkCalendarPermissionAndUpdateStateListener?: any;
```

Comment:
no better type than `any` ? last resort:

```ts
ReturnType<typeof this.props.navigation.addListener>
```

Reply:
unfortunately in this point `this` is out of class scope, I had to add 
```
ReturnType<
    NavigationScreenProp<NavigationState>["addListener"]
  >;
```